spec/support/ansible_shared/automation_manager/configuration_script.rb

```diff
@@ -85,13 +98,28 @@
   end
 
   context "creates via the API" do
-    let(:provider)      { FactoryGirl.create(:provider_ansible_tower, :with_authentication) }
+    let(:provider) do
+      FactoryGirl.create(embedded ? :provider_embedded_ansible : :provider_ansible_tower, :with_authentication)
+    end
+
     let(:manager)       { provider.managers.first }
     let(:atc)           { double("AnsibleTowerClient::Connection", :api => api) }
     let(:api)           { double("AnsibleTowerClient::Api", :job_templates => job_templates) }
     let(:job_templates) { double("AnsibleTowerClient::Collection", :create! => job_template) }
     let(:job_template)  { AnsibleTowerClient::JobTemplate.new(nil, job_template_json) }
 
+    let(:source) do
+      FactoryGirl.create(embedded ? :embedded_configuration_script_source : :ansible_configuration_script_source,
```

Comment:
I'd suggest we pass in the factory object's label. Then 3 of these if-else can be eliminated

That would make this `shared_example` agnostic of whether its an `embedded` or not. Anything specs that are not shared between the `embedded_ansible` and `ansible_tower` should be in their own spec files.